test/cpp/c10d/ProcessGroupNCCLTest.cpp

```diff
@@ -497,6 +503,36 @@ void testReduceScatter(const std::string& path, int rank, int size) {
   }
 }
 
+
+void testProcessGroupNCCLHealthCheckFail(const std::string& path, int /* unused */, int /* unused */) {
+  // simulate world_size > 1 here via threads.
+  const int worldSize = 4;
+  std::mutex m;
+  std::unordered_set<uint64_t> nums;
+  auto runTest = [&](int i) {
+    NCCLTest test(path, worldSize, std::chrono::milliseconds(3000));
+    // Catch error relating to health check failure
+    bool error_caught = false;
+    try {
+      test.initialize(0, worldSize);
```

Comment:
So the error occurs since every rank specifies rank 0?

Reply:
Yep, this is to simulate a hang which the main thread should eventually become aware of (after the 3s timeout).